internal/util/metrics/metrics.go

```diff
@@ -0,0 +1,112 @@
+package metrics
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type CustomMetrics struct {
+	pipelinesGauge          *prometheus.GaugeVec
+	pipelineCounterMap      map[string]struct{}
+	isbsvcGauge             *prometheus.GaugeVec
+	isbsvcCounterMap        map[string]struct{}
+	numaflowControllerGauge *prometheus.GaugeVec
+}
+
+var (
+	pipelineLock sync.Mutex
+	isbsvcLock   sync.Mutex
+	// the pipelineMap contains the information of all running pipeline with key as "name_namespace"
```

Comment:
```suggestion
	// the pipelineMap contains the information of all running pipelines with key as "name_namespace"
```

Reply:
Fixed, I missed to move comment after updating the code. Thanks